controller/apps.go

```diff
@@ -152,34 +152,34 @@ func (c *AppsController) getKubeClient(ctx context.Context) (kubernetes.KubeClie
 	// create Auth API client
 	authClient, err := auth.CreateClient(ctx, c.Config)
 	if err != nil {
-		log.Error(ctx, nil, "error accessing Auth server"+tostring(err))
-		return nil, errs.WithStack(err)
+		log.Error(ctx, nil, "error accessing Auth server %s", tostring(err))
+		return nil, errs.Wrapf(err, "error creating Auth client")
 	}
 
 	authUser, err := getUser(*authClient, ctx)
 	if err != nil {
-		log.Error(ctx, nil, "error accessing Auth server"+tostring(err))
-		return nil, errs.WithStack(err)
+		log.Error(ctx, nil, "error accessing Auth server: %s", tostring(err))
```

Comment:
Why are you only logging here and not in all of the other places where you return an error? Also, usually when we log, we log some values:

```go
log.Error(ctx, map[string]interface{}{
  "err": err,
}, "failed to access auth server")
```

Also, the `tostring` function seems useless. Why not call `err.Error()`? Also, I suggest to print errors with `%+v` because it dumps the stack (if available).

Reply:
I've gotten rid of the logging; it should have mostly been Debug().

Reply:
Logging is not useless. @aslakknutsen please coordinate where @stooke should put logging messages. Apparently no logging messages are in the code atm.

Reply:
@stooke If err is handled at a higher level and logged there then fine. But log to provide a trace and more info about the state of when it happened. Logging the same error and only the error in multiple locations makes no sense.

Reply:
Kwk, I don't think logging is useless.  
I've started adding properly dont' logging back.